opsdroid/connector/matrix/connector.py

```diff
@@ -104,111 +103,136 @@ def _process_rooms_dict(self, rooms):
+        response = await self.connection.sync(timeout=3000, sync_filter=first_filter_id)
+
+        if isinstance(response, nio.SyncError):
+            _LOGGER.error(
+                f"Error during initial sync: {response.message} (status code {response.status_code})"
+            )
+            return
+
+        self.connection.sync_token = response.next_batch
 
         if self.nick:
-            display_name = await self.connection.get_display_name(self.mxid)
+            display_name = await self.connection.get_displayname(self.mxid)
             if display_name != self.nick:
-                await self.connection.set_display_name(self.mxid, self.nick)
+                await self.connection.set_displayname(self.nick)
 
     async def disconnect(self):
         """Close the matrix session."""
-        await self.session.close()
+        await self.connection.close()
 
     async def _parse_sync_response(self, response):
-        self.connection.sync_token = response["next_batch"]
+        self.connection.sync_token = response.next_batch
 
         # Emit Invite events for every room in the invite list.
-        for roomid, room in response["rooms"]["invite"].items():
+        for roomid, roomInfo in response.rooms.invite.items():
             # Process the invite list to extract the person who invited us.
             invite_event = [
                 e
-                for e in room["invite_state"]["events"]
-                if "invite" == e.get("content", {}).get("membership")
+                for e in roomInfo.invite_state
+                if isinstance(e, nio.InviteMemberEvent)
+                if "invite" == e.membership
             ][0]
-            sender = await self.get_nick(None, invite_event["sender"])
+            sender = await self.get_nick(None, invite_event.sender)
 
             await self.opsdroid.parse(
                 events.UserInvite(
                     target=roomid,
-                    user_id=invite_event["sender"],
+                    user_id=invite_event.sender,
                     user=sender,
                     connector=self,
                     raw_event=invite_event,
                 )
             )
 
-        for roomid, room in response["rooms"]["join"].items():
-            if "timeline" in room:
-                for event in room["timeline"]["events"]:
-                    if event["sender"] != self.mxid:
-                        return await self._event_creator.create_event(event, roomid)
+        for roomid, roomInfo in response.rooms.join.items():
+            if roomInfo.timeline:
+                for event in roomInfo.timeline.events:
+                    if event.sender != self.mxid:
+                        if isinstance(event, nio.MegolmEvent):
+                            # event is encrypted
+                            event = await self.connection.decrypt_event(event)
+                        return await self._event_creator.create_event(
```

Comment:
I know this was like this before, but should this actually be return here? What happens if there is more than one event in the sync response?!

(we probably need to open an issue for this and not worry about it in this PR).

Reply:
I think you're right. The return seems not to be correct here

Reply:
A sorry.. no the return is needed so yeah we should probably open an issue for that